builtin/providers/alicloud/validators.go

```diff
@@ -2,26 +2,17 @@ package alicloud
 
 import (
 	"fmt"
-	"net"
-	"strconv"
+	"regexp"
 	"strings"
 
 	"github.com/denverdino/aliyungo/common"
 	"github.com/denverdino/aliyungo/ecs"
-	"github.com/denverdino/aliyungo/slb"
-	"regexp"
+	"github.com/hashicorp/terraform/helper/validation"
 )
 
 // common
 func validateInstancePort(v interface{}, k string) (ws []string, errors []error) {
-	value := v.(int)
-	if value < 1 || value > 65535 {
-		errors = append(errors, fmt.Errorf(
-			"%q must be a valid instance port between 1 and 65535",
-			k))
-		return
-	}
-	return
+	return validation.IntBetween(1, 65535)(v, k)
```

Comment:
mostly because there are a ton of tests, and I wanted to be certain that all of these worked correctly before inlining.

speaking of which, any thoughts on how to test the validation functions after I inline them? seems overkill to actually create full HCL test data for each variant.